modules/table/src/main/java/org/apache/ignite/internal/table/distributed/SortedIndexLocker.java

```diff
@@ -188,14 +154,9 @@ public CompletableFuture<Lock> locksForInsert(UUID txId, BinaryRow tableRow, Row
         // Find next key.
         Cursor<IndexRow> cursor = storage.scan(prefix, null, SortedIndexStorage.GREATER);
```

Comment:
Not related to your changes, but we should close the cursor. try-with-resources?

Reply:
good catch!